gnomad/utils/annotations.py

```diff
@@ -2388,6 +2388,70 @@ def _update_struct(
     return ht.annotate(**updated_rows)
 
 
+def fill_missing_key_combinations(
+    ht: hl.Table,
+    fill_values: Dict[str, hl.expr.Expression],
+    key_values: Optional[Dict[str, List]] = None,
+) -> hl.Table:
+    """
+    Fill missing key combinations with requested fill values.
+
+    This function fills missing key combinations in the input Table with requested fill
+    values. The fill values are specified in the `fill_values` dictionary. The unique
+    values for each key field are collected from the input Table unless specified in
+    the `key_values` dictionary.
+
+    This is useful when you want to ensure that all possible key combinations are
+    present in a Table, even if they are missing from the input Table. This can
+    happen when you are aggregating data and want to ensure that all possible key
+    combinations are present in the output Table, not only the ones that are present.
+
```

Comment:
```suggestion

    Example:

    .. code-block:: python

        ht = hl.Table.parallelize(
                [{'key1': 'A', 'key2': 1, 'value': 10},
                {'key1': 'A', 'key2': 2, 'value': 20},
                {'key1': 'B', 'key2': 1, 'value': 30}],
                hl.tstruct(key1=hl.tstr, key2=hl.tint32, value=hl.tint32),
                key=['key1', 'key2']
            )
        fill_values = {'value': hl.missing(hl.tint32)}
        filled_ht = fill_missing_key_combinations(ht, fill_values)
        filled_ht.show()

        +------+------+
        | key1 | key2 | value |
        +------+------+
        |  A   |  1   |  10   |
        |  A   |  2   |  20   |
        |  B   |  1   |  30   |
        |  B   |  2   | null  |
        +------+------+

    In this example, the input table is missing the combination (B, 2).
    After applying `fill_missing_key_combinations`, the missing key combination
    (B, 2) is filled with the specified fill value for 'value' (null in this case).

```

Reply:
I thought an example would be helpful here since its a bit abstract so I asked ChatGPT. I think this is a reasonable example.